### PR TITLE
Implement XEP-0319: Last User Interaction in Presence

### PR DESCRIPTION
**Pull request by lnjX**
https://xmpp.org/extensions/xep-0319.html

### CODE REVIEW

File: tests/qxmpppresence/tst_qxmpppresence.cpp [hunk 1]
```diff
@@ -231,5 +233,25 @@ void tst_QXmppPresence::testPresenceWithMucSupport()
...
+        "<presence to=\"coven@chat.shakespeare.lit/thirdwitch\" "
+                  "from=\"hag66@shakespeare.lit/pda\">"
+            "<idle xmlns=\"urn:xmpp:idle:1\" since=\"1969-07-21T02:56:15Z\"/>"
+        "</presence>");
+
+    QXmppPresence presence;
+    parsePacket(presence, xml);
+    QVERIFY(!presence.lastUserInteraction().isNull());
+    QVERIFY(presence.lastUserInteraction().isValid());
+    QCOMPARE(presence.lastUserInteraction(), QDateTime({1969, 7, 21}, {2, 56, 15}, Qt::UTC));
```
Comment (jlaine) [inline tests/qxmpppresence/tst_qxmpppresence.cpp]: Forgive my no-being-up-to-date-with-modern-C++, but what compilers support this syntax?

Comment (lnjX) [reply tests/qxmpppresence/tst_qxmpppresence.cpp]: Every C++14 compiler:
https://en.cppreference.com/w/cpp/language/list_initialization
https://en.cppreference.com/w/cpp/language/overload_resolution#Implicit_conversion_sequence_in_list-initialization

... but I just noticed that you you're still on C++11: https://github.com/qxmpp-project/qxmpp/blob/master/CMakeLists.txt#L12
I can change it to the old syntax if that's a problem.

Comment (jlaine) [reply tests/qxmpppresence/tst_qxmpppresence.cpp]: Yes please, let's use the old syntax. I don't really want to deal with issues from users saying "hey my build broke".

**Comment by jlaine** [comment]
Could you add XEP-0319 to doc/xep.doc please?

**Comment by lnjX** [comment]
@jlaine done :)

**Comment by jlaine** [comment]
Ah crap, needs a rebate due to the docs stuff.

Other than that looks good to me. Does anyone else have any comments?

**Comment by lnjX** [comment]
@jlaine fixed :)

**Pull request merged**